src/java/org/apache/cassandra/index/sai/QueryContext.java

```diff
@@ -77,6 +72,19 @@ public class QueryContext
     private final LongAdder shadowedKeysLoopCount = new LongAdder();
     private final NavigableSet<PrimaryKey> shadowedPrimaryKeys = new ConcurrentSkipListSet<>();
 
+    // Total count of rows in all sstables and memtables.
+    private Long totalAvailableRows = null;
+
+    // Determines the order of using indexes for filtering and sorting.
+    // Null means the query execution order hasn't been decided yet.
+    private FilterSortOrder filterSortOrder = null;
```

Comment:
As we work on exporting more metrics, I think it'd be really interesting to expose which path we take as a counter with this enum as a label.

@pkolaczk do you have any other metrics you think we should collect? It's more in the scope of https://github.com/riptano/VECTOR-SEARCH/issues/75, so we don't need to worry about it here, but it would be great to know what you think would help us to better understand how this works for real datasets.